flowey/flowey_core/src/pipeline.rs

```diff
@@ -947,6 +949,13 @@ impl PipelineJob<'_> {
         self
     }
 
+    /// Overrides the name of the job. The main use case for this is to get an
+    /// artifact in ADO with a specific name.
+    pub fn ado_override_job_name<S: Into<String>>(self, name: S) -> Self {
```

Comment:
Lets be more precise here, and call this the job ID (since that's what this string is referred to in https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/jobs-job?view=azure-pipelines)

Also, `impl AsRef<&str>` is typically preferred over `Into<String>`, since it'll accept more types, at the cost of having this code do an extra clone (which I is fine in this sort of code)